### DISCUSSION TITLE
Special characters in payload are not parsed properly

### DISCUSSION THREAD

**Discussion by jbanulso** [enisdenjo/graphql-sse#42] [Q&A] [answered]
I'm reporting the issue in this repo because after checking our subscriptions implementation across the whole stack here's where the culprit seems to be.

**Screenshot**
I had to change the actual data structure for privacy reasons, but basically this is how it looks when logging an event through `onMessage`. The characters displayed as � are characters with diacriticals such as á, é, í, ó, ú, ñ etc.
```
{
    "event": "next",
    "data": {
        "data": {
            "update": {
                "someField": {
                    "id": "1234",
                    "user": {
                        "id": "5678",
                        "fullName": "Abcdef � ��",
                        "__typename": "User"
                    },
                    "__typename": "SomeField"
                },
                "__typename": "Update"
            }
        },
        "errors": [],
        "subId": "08738bd4-57e2-4316-8d99-98ca3975100c",
        "type": "SUBSCRIPTION_DATA"
    }
}
```
**Expected Behaviour**
Special characters are encoded properly.

**Actual Behaviour**
They are not.

**Debug Information**
If you tell me how, happy to add more info.

**Further Information**
We're using the implementation with Apollo Client, as explained in the [README](https://github.com/enisdenjo/graphql-sse#apollo-client).

**Comment by enisdenjo** [comment, marked as answer]
The server is most likely not encoding the characters in UTF-8. If I pass those same characters to a subscription resolver, I get them properly decoded on the client:
```json
{
  "event": "next",
  "data": {
    "id": "2e3c1aec-3f20-483b-a0ff-0318a767ba38",
    "payload": {
      "data": {
        "chars": "á, é, í, ó, ú, ñ"
      }
    }
  }
}
```

<img width="1431" alt="Screenshot 2022-12-20 at 15 14 17" src="https://user-images.githubusercontent.com/11807600/208687221-ae9b9bb1-9a0c-49f4-9c53-286e3b1e65b3.png">

**Reply by jbanulso** [reply]
Thanks for the quick reply @enisdenjo, will double-check this 👍🏻
